src/Handler/NotFoundHandler.php

```diff
@@ -59,15 +59,14 @@ public function __construct(
      * Creates and returns a 404 response.
      *
      * @param ServerRequestInterface $request Passed to internal handler
-     * @param RequestHandlerInterface $handler Ignored.
      */
     public function handle(ServerRequestInterface $request) : ResponseInterface
     {
-        if (! $this->renderer) {
+        if ($this->renderer === null) {
             return $this->generatePlainTextResponse($request);
         }
 
-        return $this->generateTemplatedResponse($request);
+        return $this->generateTemplatedResponse($this->renderer, $request);
```

Comment:
what's the point as it is property?

Reply:
Property is defined as nullable, whereas here it is not nullable (call-time)